modules/alloydb/main.tf

```diff
@@ -0,0 +1,381 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  prefix = var.prefix == null ? "" : "${var.prefix}-"
+  # has_replicas = try(length(var.replicas) > 0, false)
+  is_regional = var.availability_type == "REGIONAL"
+  # secondary instance type is aligned with cluster type unless apply is targeting a promotion, in that
+  # case cluster will be 'primary' while instance still 'secondary'.
+  primary_cluster_name    = "${local.prefix}${var.cluster_name}"
+  primary_instance_name   = "${local.prefix}${var.name}"
+  secondary_cluster_name  = coalesce(var.secondary_cluster_name, "${var.cluster_name}-sec")
+  secondary_instance_name = coalesce(var.secondary_name, "${var.name}-sec")
+  secondary_instance_type = try(var.cross_region_replication.promote_secondary && google_alloydb_cluster.secondary[0].cluster_type == "SECONDARY" ? "SECONDARY" : google_alloydb_cluster.secondary[0].cluster_type, null)
```

Comment:
can we break this for readability on multiple lines?

Reply:
Split on multiple lines